### PR TITLE
Draft notes for our past two meetings

### PR DESCRIPTION
**Pull request by robmoss**
Apologies for the delay, here are my notes from the past two meetings:

- 11 April, discussing topics and materials for the orientation guide; and
- 9 May, TK gave a great presentation about several recent projects.

I've tried to organise each set of notes into short sections, each of which begins with a key message.

### CODE REVIEW

**Comment by robmoss** [comment]
@thaopletk please have a look at [my summary of our last meeting](https://git-is-my-lab-book.net/pr-preview/pr-71/community/meetings/2024-05-09/) and let me know if you're happy with how I've described your presentation.

**Comment by robmoss** [comment]
@EamonConway @jomonman537 here are the online previews for each meeting:

- [11 April](https://git-is-my-lab-book.net/pr-preview/pr-71/community/meetings/2024-04-11/)
- [9 May](https://git-is-my-lab-book.net/pr-preview/pr-71/community/meetings/2024-05-09/)

**Comment by robmoss** [comment]
Once we're happy with the meeting notes, I'd like to send out an email to let everyone know they're up (and if they missed TK's presentation they can get an idea of what they covered), and to reiterate some of the key messages/questions from these notes:

- We welcome presentations about research projects and experiences that relate in any way to reproducibility and good computational research practices. Presentations can be short, informal, and free-form. Please contact us if you have anything you might like to present!
- Changing our behaviour and work habits is hard, and so is finding time to develop these skills. We need to practice these skills on small problems first, rather than on large projects (and definitely not when there are tight deadlines). Should we organise an event to practice and develop specific skills on small-scale problems?
- Does anyone have any examples of code produced by ChatGPT that didn't function as intended?
- Is anyone interested in taking up Rob's offer of walking through the original code for a within-host malaria RBC model, and how restructured, tested, and documented it?

File: docs/community/meetings/2024-05-09.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
+# 9 May 2024
+
+## Presentation by T.K. Le
+
+In this meeting [T.K. Le](https://github.com/thaopletk) gave a presentation about a series of COVID-19 modelling projects and how their experiences were impacted by the choice of programming languages, model structure, editors, tools, etc.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: It should be "TK Le" (TK is kind of not an abbreviation)

File: docs/community/meetings/2024-05-09.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    We welcome presentations about research projects and experiences that relate in any way to reproducibility and good computational research practices.
+    Presentations can be short, informal, and free-form.
+
+    Please contact us if you have anything you might like to present!
+
+## Three projects, four models
+
+This work began in 2022, and was based on code that was originally written by [Eamon](https://github.com/EamonConway) and Camelia.
+T.K. adapted the code to run new scenarios across three separate modelling projects:
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: T.K. -> TK

(probably should do a Find+Replace)

File: docs/community/meetings/2024-05-09.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+1. [Modelling the impact of hybrid immunity on futute COVID-19 waves](https://doi.org/10.1186/s12879-024-09282-4);
+2. [Cost-effective boosting allocations in the post-Omicron era of COVID-19 management](https://doi.org/10.1101/2023.11.14.23298536); and
+3. Confidential scenario modelling for the Australian Department of Health & Aged Care.
+
+The workflow was divided into a series of distinct models:
+
+1. An immunological model written in R and [greta](https://greta-stats.org/);
+
+2. An agent-based model of population transmission written in C++ and R;
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: written in C++ (with post-processing in R)

File: docs/community/meetings/2024-05-09.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+3. A clinical pathways model implemented in Matlab; and
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: MATLAB (capitals)

File: docs/community/meetings/2024-05-09.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+4. A cost-effectiveness model implemented in R.
+
+T.K.'s primary activities involved implementing different vaccine schedules in the transmission model, and data visualisation of outputs from the transmission model and the clinical pathways, all of which they implemented in Python.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: TK

File: docs/community/meetings/2024-05-09.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+**Question:** Was it a benefit to have separate model implementations in different languages, with clearly defined data flows from one model to the next?
+
+Conceptually yes, but this structure also involved a lot of trade-offs — even the sheer volume of data that had to be saved by one model and read into another.
+It was difficult to pick up and use the code **as a whole**.
+And there were related issues, such as being able to successfully install [greta](https://greta-stats.org/).
+
+> Nick Tierney: I know that greta can be difficult to install, and I can help people with this.
+
+T.K. also noted that they found some minor inconsistencies between the models, such as whether a date was used to identify the start or end of its 24-hour interval.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: TK

File: docs/community/meetings/2024-05-09.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+> Eamon Conway: the worst thing about VS code is that debugging is very hard to setup.
+
+## Task management
+
+!!! tip "Key message"
+
+    Task management is hard, and switching to a new system during a project is extremely difficult.
+
+T.K. reported trying to use Gitlab issues to plan out what to do and how to do it, but found they weren't a good fit with their workflow.
+They then trialled [Trello boards](https://trello.com/) for different versions, but stopped using them due to a lack of time to manage the boards.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md:93]: singular board 😆

File: docs/community/meetings/2024-05-09.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+In review:
+
+- T.K. uses [TickTick](https://www.ticktick.com/) for personal stuff, but it's ugly!
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: I use Todoist for personal tasks (and I like it); I use TickTick for work tasks (and it's ugly)

File: docs/community/meetings/2024-05-09.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+- Trello was their favourite so far, but plotting out an entire roadmap wasn't useful — there were major changes in the plan during the project, and **it was hard to make time** to update the roadmap.
+- They are currently using TickTick and **paper-based notes**, but are always on the lookout for a better task manager.
+- One advantage of online systems are they you can easily paste hyperlinks, emails, pictures, etc.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: "...online systems **is that** you can..."

File: docs/community/meetings/2024-05-09.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+> Rob Moss: we know that behaviour changes are hard to achieve, so it's not surprising that a large change was challenging to maintain — ideally we would make small, incremental changes, but this isn't always possible or useful.
+
+> Eamon Conway: I like the general idea of using task-tracking software, but I've settled on only using paper.
+> It's always with me, it's always at home, and it's physically under my keyboard!
+
+> Ruarai Tobin: I use notion for a single large Markdown file, you can paste screenshots.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: "Notion" is capitalised

File: docs/community/meetings/2024-05-09.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+> If possible, we need to allocate sufficient time in our budgets, as best as possible.
+> Practising these skills on smaller scales can also help us to use them with less overhead in larger projects.
+
+## Version control and publishing
+
+!!! tip "Key message"
+
+    This can be challenging even with all of the tools and infrastructure in place.
+
+**Question:** were all of the projects wrapped up into one central things?
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: one central things????

File: docs/community/meetings/2024-05-09.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+
+No, they're all separate.
+The first project was provided as a zip file attached to the paper.
+The second project is in a public git repository.
+The final project is ongoing and remains confidential, it is stored in a private repository on the University of Melbourne's GitLab instance.
+
+**Question:** did the latest project build on the previous ones?
+
+Yes, and this lead to managing current and older versions of the code.
```
Comment (thaopletk) [inline docs/community/meetings/2024-05-09.md]: lead -> led

**Review by thaopletk — COMMENTED** [review]
When I tried to edit the file, it asked me to fork (????), so I've left some comments, I hope they make sense and are easy to action.

File: docs/community/meetings/2024-05-09.md (hunk 1 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Fixed. Sorry TK, I don't know why I wrote it as initials, but once I did, I did so throughout. I've resolved each instance.

File: docs/community/meetings/2024-05-09.md (hunk 2 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Done

File: docs/community/meetings/2024-05-09.md (hunk 3 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Done

File: docs/community/meetings/2024-05-09.md (hunk 4 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: I wasn't sure if MATLAB was simply a logo/emblem thing, but no, apparently it's the trade-marked name. Done.

File: docs/community/meetings/2024-05-09.md (hunk 5 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Done

File: docs/community/meetings/2024-05-09.md (hunk 8 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Thanks, my note-taking clearly couldn't keep up with the conversation. Done.

File: docs/community/meetings/2024-05-09.md (hunk 9 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Thanks for the grammar check! Done.

File: docs/community/meetings/2024-05-09.md (hunk 10 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Done, and added a link to [Notion](https://www.notion.so/).

File: docs/community/meetings/2024-05-09.md (hunk 11 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: It took me a few seconds to realise what the problem was ("things" -> "thing")! Done.

File: docs/community/meetings/2024-05-09.md (hunk 12 above)
Comment (robmoss) [reply docs/community/meetings/2024-05-09.md]: Done

**Comment by robmoss** [comment]
> When I tried to edit the file, it asked me to fork (????), so I've left some comments

We need to add you to the repository to allow you to create commits/branches without forking. I'm happy to do that, if you'd like? This would make it easier for you to propose new/modified content, if that's on your radar :)

> I hope they make sense and are easy to action.

Yes and yes. Thank you very much.

**Comment by thaopletk** [comment]
Thank you for making all the changes, Rob!

File: docs/community/meetings/2024-04-11.md [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+
+    For example, use the search bar above (press <kbd>F</kbd>) to interactively search the entire website.
+
+## Purpose of the guide
+
+We are aiming to keep the orientation guide short and simple, to avoid overwhelming newcomers.
+
+> James Ong: If we can agree on a structure, we can then get people to contribute to specific sections.
+
+> T.K. Le: schedule a one-hour meeting where everyone works on writing content for 30 minutes, and then reviews each others' content for 30 minutes?
```
Comment (thaopletk) [inline docs/community/meetings/2024-04-11.md]: omg just noticed this "T.K." 😆

Comment (robmoss) [reply docs/community/meetings/2024-04-11.md]: 🤦

Comment (robmoss) [reply docs/community/meetings/2024-04-11.md]: Fixed, and I can confirm that there are no more instances of "T.K." in the entire repository.

**Comment by jomonman537** [comment]
Thanks for sending through the meeting notes @robmoss. I found the notes very detailed, particularly TK's presentation from this month's meeting. It was interesting to see how they structured and ran their modelling projects and what they learnt from running them. These meeting notes would not only server as a useful resource for everyone, but sets everything up well for future meeting presentations. 

Going to last month's meeting notes, most of what I have written was covered. The only thing that seems to be excluded from your notes is 'recordkeeping your coding project'. Did we cover that as something to include in the orientation guide?

**Comment by robmoss** [comment]
Thanks for your review @jomonman537. Regarding record-keeping, I felt that between the "Project organisation" section (using version control) and the "Working collaboratively" section (clearly explaining the project structure and how everything works) this was already covered. Are there particular observations / items / remarks you'd like to add?

**Comment by jomonman537** [comment]
I only noted it down as a brief point under my notes. But I think my point would be covered by the two sections that you mentioned. 
Other than that, I have nothing to add from my end. Great work with your notes!

**Comment by robmoss** [comment]
Thanks very much @jomonman537.

@EamonConway please have a look when you have the time :)

**Comment by EamonConway** [comment]
Happy to have this merged. 
A great summary of two good meetings!

**Comment by robmoss** [comment]
Thanks a tonne @EamonConway, @jomonman537, @thaopletk :)

**Pull request merged**